### PR TITLE
Small cleanups for the new subtraction operator

### LINKED ISSUE #1377: Small cleanups for the new subtraction operator
Address post merge feedback in https://github.com/aio-libs/yarl/pull/1340

https://github.com/aio-libs/yarl/blob/04d853c1e87df295981b9d2833c09b2b485a1800/yarl/_path.py#L60

### PR DESCRIPTION
**Pull request by oleksbabieiev**
## What do these changes do?

Small cleanups for the new subtraction operator.

## Are there changes in behavior for the user?

N/A

## Related issue number

Resolves #1377 

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes

### CODE REVIEW

File: tests/test_url.py [hunk 1]
```diff
@@ -85,29 +85,36 @@ def test_sub(target: str, base: str, expected: str):
...
+    expected_error_msg = r"^'..' segment in '/path/..' cannot be walked$"
     with pytest.raises(ValueError, match=expected_error_msg):
         URL("path/to") - URL("/path/../from")
 
 
+def test_sub_combined_with_join():
+    target = URL("http://example.com/path/to")
+    base = URL("http://example.com/")
+    rel = target - base
+    assert target == base.join(rel)
```
Comment (webknjaz) [inline tests/test_url.py:115]:
```suggestion
    assert target == base / rel
```

File: tests/test_url.py [hunk 2]
```diff
@@ -85,29 +85,36 @@ def test_sub(target: str, base: str, expected: str):
...
 
 
 def test_sub_with_two_dots_in_base():
-    expected_error_msg = "'..' segment in '/path/..' cannot be walked"
+    expected_error_msg = r"^'..' segment in '/path/..' cannot be walked$"
     with pytest.raises(ValueError, match=expected_error_msg):
         URL("path/to") - URL("/path/../from")
 
 
+def test_sub_combined_with_join():
```
Comment (webknjaz) [inline tests/test_url.py:111]: Can we make this hypothesis-based?

File: tests/test_url.py (hunk 1 above)
Comment (oleksbabieiev) [reply tests/test_url.py:115]: `URL.join(url)` and `/` are not interchangeable 😅;
`/` is the equivalent of `URL.joinpath(other, encoded=False)`;
So if you want to use `/`, you need to write `base / rel.path`.

Comment (oleksbabieiev) [reply tests/test_url.py:121]: But even `base / rel.path` is not the same as `base.join(rel)`

**Comment by oleksbabieiev** [comment]
I understood why some tests fail. I will try to do something about it.

**Comment by bdraco** [comment]
```
FAILED tests/test_url.py::test_sub_combined_with_join - hypothesis.errors.FailedHealthCheck: Data generation is extremely slow: Only produced 9 valid examples in 1.05 seconds (0 invalid ones and 7 exceeded maximum size). Try decreasing size of the data you're generating (with e.g. max_size or max_leaves parameters).
```
It looks like its failing on performance.  The performance is nearly quadratic unfortunately so any long paths are going to be very slow.

**Comment by bdraco** [comment]
I tried to solve the performance issues in https://github.com/aio-libs/yarl/pull/1388, however in the process I became aware that there are other considerations that are missing.

Its clear this isn't ready to go yet and I merged #1340 too soon (apologies to everyone involved) so I've opened up https://github.com/aio-libs/yarl/pull/1391 to revert it so we can start over with a new approach.

I'm not sure if #1388 is the way to go or not but if it is, its probably a good start.

File: tests/test_url.py (hunk 1 above)
Comment (webknjaz) [reply tests/test_url.py]: Ah, I see.. This was my attempt to suggest a proper round-trip data conversion so it could be made into a property-based test.

**Comment by webknjaz** [comment]
Side note: it would probably be useful to have own hypothesis strategy under `yarl.testing` that could be utilized both within yarl's own test suite and any downstream/end-user projects. That strategy would need to produce various URL edge cases.
This is not in the scope of this PR, however, I just wanted to record the idea.

**Comment by oleksbabieiev** [comment]
Well, I think it would make sense if I create a new pull request to rework this feature @bdraco

**Comment by oleksbabieiev** [comment]
#1392
